### DISCUSSION TITLE
Does next.js incremental static regeneration serverless functions count in the `Serverless Functions Created per Month` limit?

### DISCUSSION THREAD

**Discussion by shunkakinoki** [vercel/vercel#5093] [Help] [answered]
Hi, Vercel

First of all, I wanted to say that I really love your work. 
next.js, swr, hyper, they truly are magical. 
You've earned my love for the brand.

I currently use the blazing-edge next.js v9.5 ISR feature running on vercel. 
I absolutely love it. 
It is soooooo fast, and it feels like the way it should be in 2020.

I was wondering if you could clear out the meaning behind `Serverless Functions Created per Month`
Does the monthly limit include next.js static generation serverless functions or not?

On the limit page, it writes
```NOTE: Next.js Static Site Generation and Next.js Preview Serverless Functions are excluded from the count for "Serverless Functions per Deployment".```
but is it not applied to `per month`, not just `per deployment`?

**Comment by samcx** [comment]
Hello,

Next.js SSG (Static Site Generation) does not create serverless functions (Lambdas λ) unless using Preview Mode. For more information about `Serverless Functions Created per Month` → https://vercel.com/docs/platform/limits#serverless-functions-per-month-(hobby).

Since they are not included from the per Deployment count limit, they will also not be applied to the per month limit.

**Comment by shunkakinoki** [comment]
Hi @samsisle 

Thanks for the answer.

How about next.js ISR (Incremental Static Regeneration)?

**Reply by samcx** [reply]
Hello,

iSR will create serverless functions. One thing to note — in Vercel (a recent update), Next.js SSR pages and pages/api routes get bundled into two separate Lambdas (λ), so you should not have to worry about the Serverless Function limit. I am not sure how this behavior works with the limit per month. I will relay this to the Vercel team and get back to you.

**Comment by valse** [comment, marked as answer]
Hi, the question here is wrong: the problem is not if ISR count in the monthly limit as number but as time consuming.

I have a SSG site powered by ISR for a lot of pages: I have a daily average of 30K page views and my serverless functions execution limit is very high even if I have no pages with a revalidate time under the 180 seconds... and far from the 1 second revalidate time for keep the pages updated.

With an amount of traffic and time for regeneration like this you must expect to pay for the ISR service that in my opinion is a great service but extremely not free of charge.

The real problem here is the revalidate time used for "unmutable" pages were the function use time to check possible updates even if not needed... And your time limit increase.

I'm sure that will be optimizations and new implementations for ISR (like by APIs request) where you'll have more control... but in meantime you must pay attention to use this feature: with great power there must also come great responsibility 😉
